### ISSUE TITLE
Restoring add-ons from full backup failed partially

### ISSUE THREAD

**Issue by codyc1515** [home-assistant/supervisor#4814]
### Describe the issue you are experiencing

My add-ons did not restore when I performed a complete restore. This happened across multiple attempts.

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Steps to reproduce the issue

1. Use Raspberry Pi Imager to install image to USB SSD
2. Boot
3. Restore from full backup

### Anything in the Supervisor logs that might be useful for us?

```txt
volumes = [RE_VOLUME.match(entry) for entry in config.get(ATTR_MAP, [])]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/addons/validate.py", line 270, in <listcomp>
    volumes = [RE_VOLUME.match(entry) for entry in config.get(ATTR_MAP, [])]
               ^^^^^^^^^^^^^^^^^^^^^^
TypeError: expected string or bytes-like object, got 'dict'[0m
[33m24-01-12 20:57:04 WARNING (MainThread) [supervisor.backups.backup] Can't restore Add-on core_ssh: [0m
[32m24-01-12 20:57:04 INFO (MainThread) [supervisor.addons.addon] Restore config for addon core_silabs_multiprotocol[0m
[32m24-01-12 20:57:04 INFO (MainThread) [supervisor.addons.addon] Restore/Install of image for addon core_silabs_multiprotocol[0m
[32m24-01-12 20:57:04 INFO (MainThread) [supervisor.docker.interface] Downloading docker image homeassistant/aarch64-addon-silabs-multiprotocol with tag 2.4.3.[0m
/usr/src/supervisor/supervisor/addons/addon.py:1214: RuntimeWarning: coroutine 'Addon._check_ingress_port' was never awaited
  self._check_ingress_port()
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
[32m24-01-12 20:57:47 INFO (MainThread) [supervisor.addons.addon] Restoring data and config for addon core_silabs_multiprotocol[0m
[32m24-01-12 20:57:47 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/aarch64-addon-silabs-multiprotocol with version 2.4.3[0m
[32m24-01-12 20:57:50 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/aarch64-addon-silabs-multiprotocol with version 2.4.3[0m
[32m24-01-12 20:57:50 INFO (MainThread) [supervisor.addons.addon] Finished restore for add-on core_silabs_multiprotocol[0m
[32m24-01-12 20:57:50 INFO (MainThread) [supervisor.addons.manager] Detect new Add-on after restore core_silabs_multiprotocol[0m
[31m24-01-12 20:57:50 ERROR (MainThread) [supervisor.jobs] Unhandled exception: expected string or bytes-like object, got 'dict'
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 289, in wrapper
    return await self._method(obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 1171, in restore
    data = SCHEMA_ADDON_BACKUP(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/voluptuous/schema_builder.py", line 287, in __call__
    return self._compiled([], data)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/voluptuous/schema_builder.py", line 611, in validate_dict
    return base_validate(path, iteritems(data), out)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/voluptuous/schema_builder.py", line 401, in validate_mapping
    cval = cvalue(key_path, value)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/voluptuous/validators.py", line 234, in _run
    return self._exec(self._compiled, value, path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/voluptuous/validators.py", line 361, in _exec
    v = func(path, v)
        ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/voluptuous/schema_builder.py", line 834, in validate_callable
    return schema(data)
           ^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/addons/validate.py", line 270, in _migrate
    volumes = [RE_VOLUME.match(entry) for entry in config.get(ATTR_MAP, [])]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/addons/validate.py", line 270, in <listcomp>
    volumes = [RE_VOLUME.match(entry) for entry in config.get(ATTR_MAP, [])]
               ^^^^^^^^^^^^^^^^^^^^^^
TypeError: expected string or bytes-like object, got 'dict'[0m
[33m24-01-12 20:57:50 WARNING (MainThread) [supervisor.backups.backup] Can't restore Add-on core_samba: [0m
[32m24-01-12 20:57:53 INFO (MainThread) [supervisor.addons.addon] Restore config for addon core_mariadb[0m
[32m24-01-12 20:57:53 INFO (MainThread) [supervisor.addons.addon] Restore/Install of image for addon core_mariadb[0m
[32m24-01-12 20:57:53 INFO (MainThread) [supervisor.docker.interface] Downloading docker image homeassistant/aarch64-addon-mariadb with tag 2.6.1.[0m
[32m24-01-12 20:58:01 INFO (MainThread) [supervisor.api.middleware.security] /network/info access from core_silabs_multiprotocol[0m
[32m24-01-12 20:58:05 INFO (MainThread) [supervisor.discovery] Sending discovery to Home Assistant otbr from core_silabs_multiprotocol[0m
[32m24-01-12 20:58:05 INFO (MainThread) [supervisor.discovery] Discovery 639e97c100644b67b266f8cd407bc282 message ignore[0m
[32m24-01-12 20:58:21 INFO (MainThread) [supervisor.addons.addon] Restoring data and config for addon core_mariadb[0m
[32m24-01-12 20:58:22 INFO (MainThread) [supervisor.docker.interface] Attaching to homeassistant/aarch64-addon-mariadb with version 2.6.1[0m
[32m24-01-12 20:58:28 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on homeassistant/aarch64-addon-mariadb with version 2.6.1[0m
[32m24-01-12 20:58:28 INFO (MainThread) [supervisor.addons.addon] Finished restore for add-on core_mariadb[0m
[32m24-01-12 20:58:28 INFO (MainThread) [supervisor.addons.manager] Detect new Add-on after restore core_mariadb[0m
[32m24-01-12 20:58:28 INFO (MainThread) [supervisor.backups.manager] Restore be7a8382 starting stage await_home_assistant_restart[0m
[32m24-01-12 20:58:28 INFO (MainThread) [supervisor.backups.manager] Restore be7a8382 starting stage await_addon_restarts[0m
[33m24-01-12 20:58:32 WARNING (MainThread) [supervisor.services.modules.mysql] Can't remove not exists services[0m
[32m24-01-12 20:58:34 INFO (MainThread) [supervisor.services.modules.mysql] Set core_mariadb as service provider for MySQL[0m
[33m24-01-12 20:59:50 WARNING (MainThread) [supervisor.addons.addon] Timeout while waiting for addon Silicon Labs Multiprotocol to start, took more than 120 seconds[0m
[33m24-01-12 21:00:28 WARNING (MainThread) [supervisor.addons.addon] Timeout while waiting for addon MariaDB to start, took more than 120 seconds[0m
[32m24-01-12 21:00:28 INFO (MainThread) [supervisor.backups.manager] Restore be7a8382 starting stage check_home_assistant[0m
[32m24-01-12 21:00:28 INFO (SyncWorker_0) [supervisor.docker.manager] Starting homeassistant[0m
[32m24-01-12 21:00:29 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready[0m
[32m24-01-12 21:00:37 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state freeze[0m
[32m24-01-12 21:00:37 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete[0m
[32m24-01-12 21:00:38 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/esphome/home-assistant-addon repository[0m
[32m24-01-12 21:00:38 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository[0m
[32m24-01-12 21:00:38 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/addons repository[0m
[32m24-01-12 21:00:38 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/adamoutler/HassOSConfigurator repository[0m
[32m24-01-12 21:00:38 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/alexbelgium/hassio-addons repository[0m
[32m24-01-12 21:00:38 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/koush/scrypted repository[0m
[32m24-01-12 21:00:39 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json[0m
[33m24-01-12 21:00:42 WARNING (SyncWorker_2) [supervisor.store.data] Can't read /data/addons/git/09e60fb6/external/ring-client-api/packages/homebridge-ring/config.schema.json: Invalid Add-on config!. Got {'pluginAlias': 'Ring', 'pluginType': 'platform', 'singular': True, 'customUi': True, 'customUiPath': './lib/homebridge-ui', 'customUiDevServer': 'http://localhost:3000/api/plugins/settings-ui/homebridge-ring', 'headerDisplay': 'See [the wiki](https://github.com/dgreif/ring/wiki/Refresh-Tokens) for details on generating a refresh token', 'accountLinkingHeader': 'Please supply your Ring credentials.  These will be used to contact the Ring servers and generate a unique access token for your acc...[0m
[33m24-01-12 21:00:43 WARNING (SyncWorker_0) [supervisor.store.data] Can't read /data/addons/git/09e60fb6/external/ring-client-api/packages/homebridge-ring/config.schema.json: Invalid Add-on config!. Got {'pluginAlias': 'Ring', 'pluginType': 'platform', 'singular': True, 'customUi': True, 'customUiPath': './lib/homebridge-ui', 'customUiDevServer': 'http://localhost:3000/api/plugins/settings-ui/homebridge-ring', 'headerDisplay': 'See [the wiki](https://github.com/dgreif/ring/wiki/Refresh-Tokens) for details on generating a refresh token', 'accountLinkingHeader': 'Please supply your Ring credentials.  These will be used to contact the Ring servers and generate a unique access token for your acc...[0m
[32m24-01-12 21:00:43 INFO (MainThread) [supervisor.store] Loading add-ons from store: 160 all - 0 new - 0 remove[0m
[32m24-01-12 21:00:43 INFO (MainThread) [supervisor.store] Loading add-ons from store: 160 all - 0 new - 0 remove[0m
[32m24-01-12 21:00:45 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m24-01-12 21:00:45 INFO (MainThread) [supervisor.homeassistant.core] Home Assistant Core state changed to NOT_RUNNING[0m
[33m24-01-12 21:02:58 WARNING (MainThread) [supervisor.addons.addon] core_silabs_multiprotocol is already running![0m
[32m24-01-12 21:03:56 INFO (MainThread) [supervisor.homeassistant.core] Home Assistant Core state changed to RUNNING[0m
[32m24-01-12 21:03:56 INFO (MainThread) [supervisor.homeassistant.core] Detect a running Home Assistant instance[0m
[32m24-01-12 21:03:56 INFO (MainThread) [supervisor.backups.manager] Full-Restore be7a8382 done[0m
[33m24-01-12 21:03:57 WARNING (MainThread) [supervisor.addons.addon] core_silabs_multiprotocol is already running![0m
[33m24-01-12 21:04:58 WARNING (MainThread) [supervisor.addons.addon] Timeout while waiting for addon Silicon Labs Multiprotocol to start, took more than 120 seconds[0m
[33m24-01-12 21:05:07 WARNING (MainThread) [supervisor.addons.addon] core_silabs_multiprotocol is already running![0m
[33m24-01-12 21:05:57 WARNING (MainThread) [supervisor.addons.addon] Timeout while waiting for addon Silicon Labs Multiprotocol to start, took more than 120 seconds[0m
[33m24-01-12 21:06:29 WARNING (MainThread) [supervisor.addons.addon] core_silabs_multiprotocol is already running![0m
[33m24-01-12 21:07:07 WARNING (MainThread) [supervisor.addons.addon] Timeout while waiting for addon Silicon Labs Multiprotocol to start, took more than 120 seconds[0m
[33m24-01-12 21:07:31 WARNING (MainThread) [supervisor.addons.addon] core_silabs_multiprotocol is already running![0m
[33m24-01-12 21:07:42 WARNING (MainThread) [supervisor.addons.addon] core_mariadb is already running![0m
[33m24-01-12 21:08:10 WARNING (MainThread) [supervisor.addons.addon] core_silabs_multiprotocol is already running![0m
```

## System Information

version | core-2024.1.2
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.6
os_name | Linux
os_version | 6.1.63-haos-raspi
arch | aarch64
timezone | Pacific/Auckland
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.33.0
Stage | running
Available Repositories | 1381
Downloaded Repositories | 19
HACS Data | ok

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 11.4
-- | --
update_channel | stable
supervisor_version | supervisor-2023.12.0
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 109.3 GB
disk_used | 5.3 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | Silicon Labs Multiprotocol (2.4.3), MariaDB (2.6.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 2
-- | --
resources | 1
views | 1
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | January 10, 2024 at 23:52
-- | --
current_recorder_run | January 12, 2024 at 21:13
estimated_db_size | 21.78 MiB
database_engine | mysql
database_version | 10.6.12

</details>

<details><summary>Xiaomi Miot Auto</summary>

component_version | 0.7.15
-- | --
can_reach_server | ok
can_reach_spec | ok
logged_accounts | 1
total_devices | 7

</details>

### Supervisor diagnostics

[config_entry-hassio-2a41fa358d35c5c2944924b8fd9a7f2c.json.txt](https://github.com/home-assistant/supervisor/files/13916424/config_entry-hassio-2a41fa358d35c5c2944924b8fd9a7f2c.json.txt)

**Comment by codyc1515** [comment]
Maybe was fixed in https://github.com/home-assistant/supervisor/commit/e08c8ca26db9e924fb6825bd7c0c2f39fbdd28f0

**Comment by codyc1515** [comment]
Yes, this was caused by a versioning issue on my side with the supervisor.

**Comment by agners** [comment]
Fixed by #5111.
